CMakeLists.txt

```diff
@@ -449,28 +451,63 @@ endif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 add_subdirectory(external_libraries/gidpost)
 add_subdirectory(kratos)
 
-# Configure kratos applications
-message("-- Configuring applications (ENV):")
-set_property(GLOBAL PROPERTY LIST_OF_APPLICATIONS_ADDED_THROUGH_DEPENDENCIES)
-if(DEFINED ENV{KRATOS_APPLICATIONS})
-    foreach(APPLICATION_PATH $ENV{KRATOS_APPLICATIONS})
-        get_filename_component(APPLICATION_NAME ${APPLICATION_PATH} NAME)
-        list(APPEND LIST_OF_APPLICATIONS_TO_BE_COMPILED ${APPLICATION_NAME})
-    endforeach(APPLICATION_PATH $ENV{KRATOS_APPLICATIONS})
-endif(DEFINED ENV{KRATOS_APPLICATIONS})
+# Collect Kratos applications for the CMake GUI / CMake variables
+set(KRATOS_APPLICATION_DIRECTORIES "${KRATOS_SOURCE_DIR}/applications" CACHE STRING "List of directories with Kratos applications")
```

Comment:
TL;DR: I don't see why it wouldn't.

```KRATOS_APPLICATION_DIRECTORIES``` just tells CMake where to look for applications. Subdirectories within each entry of ```KRATOS_APPLICATION_DIRECTORIES``` are globbed afterwards, and are assumed to be kratos applications (at least they should contain a *CMakeLists.txt*). The way you'd use is by appending the parent directory of your out-of-tree applications to it.

For example, if you have the following directory structure:
```
.
└─ kratos
│    └─  applications
│    │    └─ FluidDynamicsApplication
│    │    │   │    CMakeLists.txt
│    │    │   │    ...
│    │    └─ ...
│    └─  ...
└─ external_kratos_applications
│    └─ MyExternalApplication_1
│    │    │    CMakeLists.txt
│    │    │    ...
```
just append ```./external_kratos_applications``` to ```KRATOS_APPLICATION_DIRECTORIES```. If you decide to add another external application ```MyExternalApplication_2```, it will get discovered without modifying anything (after rerunning CMake of course).
```
.
└─ kratos
│    └─  applications
│    │    └─ FluidDynamicsApplication
│    │    │   │    CMakeLists.txt
│    │    │   │    ...
│    │    └─ ...
│    └─  ...
└─ external_kratos_applications
│    └─ MyExternalApplication_1
│    │    │    CMakeLists.txt
│    │    │    ...
│    └─ MyExternalApplication_2
│    │    │    CMakeLists.txt
│    │    │    ...
```
Naturally, you'd still have to explicitly specify that you want to build ```MyExternalApplication_2``` too.